### ISSUE TITLE
EIP1559/legacy artifacts

### ISSUE THREAD

**Issue by odyslam** [foundry-rs/foundry#3598]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (70f4fb5 2022-11-02T00:05:06.592613Z)

### What command(s) is the bug in?

forge script

### Operating System

macOS (Apple Silicon)

### Describe the bug

When developing `parseJson` helpers to automatically parse forge-script artifacts:

> Seeing some artifacts from a chain that I know doesn't support EIP1559 and it has the gas field without the gasPrice, as you would expect from a LegacyTransaction. It could be that their RPC returns the info incorrectly, so there is nothing forge can do. 

Basically, I deployed to MilkomedaC1, which is not EIP1559 but got EIP1559-type artifacts.

**Comment by mds1** [comment]
@odyslam What exactly is the issue here? This sounds like an issue with the MilkomedaC1 RPC and not forge?

**Comment by odyslam** [comment]
@mds1, if I remember correctly, for some reason, several nonERC1559 chains produced ERC1559 artifacts. I will need to investigate again.

**Comment by zerosnacks** [comment]
Marking as `stale`, this has likely been resolved over time. If this is not the case feel free to re-open.
